File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-swl/core/core/exchanger/impl/SwlExpireCacheNonceManager.js
import SwlNonceManager from "../SwlNonceManager";
import LocalStorageExpireCache from "../../../../../i2f-core/cache/impl/LocalStorageExpireCache";

/**
 * @return {SwlExpireCacheNonceManager}
 * @implements {SwlNonceManager}
 * @constructor {SwlExpireCacheNonceManager}
 */
function SwlExpireCacheNonceManager() {
    /**
     *
     * @type {IExpireCache}
     */
    this.cache = new LocalStorageExpireCache();
    /**
     *
     * @type {string}
     */
    this.cacheKeyPrefix=null;
}

// 继承
SwlExpireCacheNonceManager.prototype = Object.create(SwlNonceManager.prototype)
SwlExpireCacheNonceManager.prototype.constructor = SwlExpireCacheNonceManager

SwlExpireCacheNonceManager.NONCE_PREFIX_KEY=function(){
    return "swl:nonce:";
}

/**
 *
 * @param key {String}
 * @return {String}
 */
SwlExpireCacheNonceManager.prototype.cacheKey=function(key) {
    let cacheKeyPrefix = this.cacheKeyPrefix;
    if (cacheKeyPrefix == null || cacheKeyPrefix.length==0) {
        return key;
    }
    return cacheKeyPrefix + ":" + key;
}

/**
 *
 * @param nonce {String}
 * @return {String}
 */
SwlExpireCacheNonceManager.prototype.nonceKey=function(nonce) {
    return SwlExpireCacheNonceManager.NONCE_PREFIX_KEY() + nonce;
}

/**
 * @param nonce {String}
 * @return {boolean}
 */
SwlExpireCacheNonceManager.prototype.contains = function (nonce) {
    return this.cache.exists(this.cacheKey(this.nonceKey(nonce)));
}
/**
 * @param nonce {String}
 * @param timeoutSeconds {long}
 * @return {void}
 */
SwlExpireCacheNonceManager.prototype.set = function (nonce,timeoutSeconds) {
    this.cache.setWith(this.cacheKey(this.nonceKey(nonce)),nonce,timeoutSeconds);
}

export default SwlExpireCacheNonceManager

File: i2f-jdk/i2f-swl/src/main/java/i2f/swl/core/exchanger/impl/SwlExpireCacheNonceManager.java
package i2f.swl.core.exchanger.impl;

import i2f.cache.impl.container.MapCache;
import i2f.cache.impl.expire.ObjectExpireCacheWrapper;
import i2f.cache.std.expire.IExpireCache;
import i2f.swl.core.exchanger.SwlNonceManager;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.TimeUnit;

/**
 * @author Ice2Faith
 * @date 2026/4/9 10:39
 * @desc
 */
@Data
@NoArgsConstructor
public class SwlExpireCacheNonceManager implements SwlNonceManager {
    public static final String NONCE_PREFIX_KEY = "swl:nonce:";

    private IExpireCache<String, String> cache = new ObjectExpireCacheWrapper<>(new MapCache<>(new ConcurrentHashMap<>()));
    private String cacheKeyPrefix;

    public String cacheKey(String key) {
        String cacheKeyPrefix = this.cacheKeyPrefix;
        if (cacheKeyPrefix == null || cacheKeyPrefix.isEmpty()) {
            return key;
        }
        return cacheKeyPrefix + ":" + key;
    }

    public String nonceKey(String nonce) {
        return NONCE_PREFIX_KEY + nonce;
    }

    @Override
    public boolean contains(String nonce) {
        return cache.exists(cacheKey(nonceKey(nonce)));
    }

    @Override
    public void set(String nonce, long timeoutSeconds) {
        cache.set(cacheKey(nonceKey(nonce)),nonce,timeoutSeconds, TimeUnit.SECONDS);
    }
}
